### PR TITLE
ARC: Add mpuv8 support

### PR DESCRIPTION
**Pull request by kokas-a**
Turn on mpuv8 support for ARC SEM cpus

### CODE REVIEW

File: boards/arc/nsim/support/nsim_sem.props [hunk 1]
```diff
@@ -30,7 +30,7 @@
 	nsim_isa_smart_stack_entries=8
 	mpu_sid_option=1
 	mpu_regions=16
-	mpu_version=4
```
Comment (evgeniy-paltsev) [inline boards/arc/nsim/support/nsim_sem.props:33]: We don't want to modify existing nsim_sem board as we till going to support MPUv4. Moreover, by bumping MPU version without syncing everything else with latest SEM template(s) we introduce the configuration which may exist only in simulation but not in real HW.

As we don't introduce any new code to handle MPUv8 separately I guess it's fine to keep it untested for now.

File: arch/arc/core/mpu/Kconfig [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 config ARC_MPU_VER
 	int "ARC MPU version"
-	range 2 6
+	range 2 8
```
Comment (evgeniy-paltsev) [inline arch/arc/core/mpu/Kconfig:8]: Now we implicitly introduce MPUv7 support here.

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

File: arch/arc/core/mpu/Kconfig [hunk 3]
```diff
@@ -5,12 +5,12 @@
...
 	default 2
 	help
 	  ARC MPU has several versions. For MPU v2, the minimum region is 2048 bytes;
 	  For other versions, the minimum region is 32 bytes; v4 has secure features,
-	  v6 supports up to 32 regions.
+	  v6 supports up to 32 regions. Note: MPU v5 & v7 not supported.
```
Comment (evgeniy-paltsev) [inline arch/arc/core/mpu/Kconfig:13]:
```suggestion
	  v6 supports up to 32 regions. Note: MPU v5 & v7 are not supported.
```

File: soc/arc/snps_nsim/Kconfig.defconfig.sem [hunk 4]
```diff
@@ -16,7 +16,7 @@ config NUM_IRQS
 	default 30
 
 config ARC_MPU_VER
-	default 4
+	default 8
```
Comment (evgeniy-paltsev) [inline soc/arc/snps_nsim/Kconfig.defconfig.sem:19]: As we've dropped nsim props / mdb args changes we also need to drop this change.

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

**Review by abrodkin — APPROVED** [review]

**Review by evgeniy-paltsev — APPROVED** [review]

**Pull request merged**